### PR TITLE
RHOARDOC-1021 JVM metrics using Jolokia

### PR DESCRIPTION
**Pull request by inoxx03**
procedure for accessing the hawt.io JVM console on Minishift

### CODE REVIEW

File: docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[#accessing-jvm-metrics-using-jolokia]
+= Accessing JVM metrics using Jolokia on OpenShift
+
+link:https://jolokia.org/documentation.html[Jolokia^] is a built-in lightweight solution for accessing JMX (Java Management Extension) metrics over HTTP on OpenShift.
+It allows you to access CPU, storage, and memory usage data collected by JMX over an HTTP bridge using a REST interface and JSON-formatted message payloads, making it suitable for monitoring cloud applications, due to its comparably high speed and low resource-requirements.
+
+
+All Fabric8 container images expose port `8778` by default on the host of your {OpenShiftLocal}.
```
Comment (Ladicek) [inline docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: How are Fabric8 images relevant in this context?

File: docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+= Accessing Jolokia metrics using the hawt.io console
+
+For Java-based applications, the OpenShift Web console provides the integrated link:https://docs.openshift.com/container-platform/3.6/architecture/infrastructure_components/web_console.html#jvm-console[_hawt.io_ console^] that collects and displays all relevant metrics output by the JVM running your application.
+
+.Prerequistes
+
+* the `oc` client uthenticated
+* a Java-based application container running in a project on OpenShift
+* port `8778` exposed in the container image (exposed by default in Fabric8 application images)
+* latest link:https://github.com/jboss-container-images/redhat-openjdk-18-openshift-image/blob/openjdk18-dev/image.yaml[JDK 1.8.0 image^]
```
Comment (Ladicek) [inline docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: Maybe don't mention the Fabric8 images at all and instead explain how to enable it on _this_ image?

File: docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc (hunk 1 above)
Comment (inoxx03) [reply docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: @Ladicek I see, they aren't really:) I will remove the references to Fabric8 images form this section.

**Comment by inoxx03** [comment]
@Ladicek  I removed the references to the Fabric8 images and added a steps to edit the yaml template to expose port `8778` and redeploy the application pod. WDYT?
(IIRC, exposing a service on port 8778 using `oc expose` will not work in this context, but i might be missing something)

File: docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc [hunk 3]
```diff
@@ -4,27 +4,68 @@
...
 * latest link:https://github.com/jboss-container-images/redhat-openjdk-18-openshift-image/blob/openjdk18-dev/image.yaml[JDK 1.8.0 image^]
 
 .Procedure
 
-. Log into the OpenShift Web console.
+. Obtain the name of the pod running your application:
++
+[source,bash,options="nowrap",subs="attributes+"]
+--
+oc get pods
```
Comment (Ladicek) [inline docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: So if you below edit the deployment config, shouldn't you list the deployment configs here? `oc get dc`

Comment (inoxx03) [reply docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: @Ladicek, yes, that should by all means be mentioned. I'll add it in, thank you for catching this:)

Comment (inoxx03) [reply docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: @Ladicek I've added the step to list the deployment config using `oc get dc`

Comment (Ladicek) [reply docs/topics/proc_accessing-jolokia-using-hawtio-on-openshift.adoc]: I think you don't really need to look for the pods. After changing the dc, the pods will have different names anyway. I'd personally get rid of `oc get pods` and just did `oc get dc` (_not_ `oc get dc app-name`). That will tell me the names of all deployment configs and from there, I select the one which I will `oc edit`.

It's not that big of a deal really -- pods managed by a dc are always named after the dc. Your call :-)

**Comment by inoxx03** [comment]
@Ladicek thank you for the feedback, I incorporated your suggestions. Please let me know what you think:)

File: docs/spring-boot-runtime/master.adoc [hunk 4]
```diff
@@ -15,6 +15,15 @@ include::topics/sb-product-versions.adoc[leveloffset=+2]
 
 include::topics/sb-features-and-frameworks-summary.adoc[leveloffset=+2]
 
+== Monitoring
+
+This section contains information about monitoring your {SpringBoot}&ndash;based application running on OpenShift.
```
Comment (inoxx03) [inline docs/spring-boot-runtime/master.adoc]: a. remove double space before variable
b. AFAIK, the `&ndash;` HTML entity is somehow causing the checks to fail

**Comment by inoxx03** [comment]
@rhoads-zach Fixed the build and updated PR.
Had to temporarily disable validation because of hard-coded HTML entiites.
Please feel free to merge:)

**Comment by rhoads-zach** [comment]
LGTM, merging.

**Pull request merged**